partiql-eval/src/eval/expr/operators.rs

```diff
@@ -0,0 +1,327 @@
+                            // - https://github.com/partiql/partiql-lang-kotlin/pull/621#issuecomment-1147754213
+
+                            // TODO change the Null propagation if required.
+                            // Current implementation propagates `Null` as described in PartiQL spec section 8 [1]
+                            // and differs from `partiql-lang-kotlin` impl [2].
+                            // [1] https://github.com/partiql/partiql-lang-kotlin/issues/896
+                            // [2] https://partiql.org/assets/PartiQL-Specification.pdf#section.8
+                            Some(mut iter) => {
+                                let ret = iter
+                                    // init accum to `false`, look for the value
+                                    .try_fold(Boolean(false), |accum, elem| match elem {
+                                        // found it, short circuit and immediately return `true`
+                                        x if x == lhs => ControlFlow::Break(Boolean(true)),
+                                        // found null/missing; return `Null` if `lhs` is not subsequently found
+                                        Null | Missing => ControlFlow::Continue(Null),
+                                        // something else; carry forward knowledge of previous null/missing and keep looking
+                                        _ => ControlFlow::Continue(accum),
+                                    });
+                                match ret {
+                                    ControlFlow::Continue(v) => v,
+                                    ControlFlow::Break(v) => v,
+                                }
+                            }
+                            None => Null,
+                        }
+                    }
+                )
+            }
+            EvalOpBinary::Concat => {
+                create!(Check<STRICT>, [TYPE_ANY, TYPE_ANY], |lhs, rhs| {
+                    // TODO non-naive concat (i.e., don't just use debug print for non-strings).
+                    let lhs = if let Value::String(s) = lhs {
+                        s.as_ref().clone()
+                    } else {
+                        format!("{lhs:?}")
+                    };
+                    let rhs = if let Value::String(s) = rhs {
+                        s.as_ref().clone()
+                    } else {
+                        format!("{rhs:?}")
+                    };
+                    Value::String(Box::new(format!("{lhs}{rhs}")))
+                })
+            }
+        }
+    }
+}
+
+/// Represents an evaluation PartiQL `BETWEEN` operator, e.g. `x BETWEEN 10 AND 20`.
+#[derive(Debug, Default, Clone)]
+pub(crate) struct EvalBetweenExpr {}
+
+impl BindEvalExpr for EvalBetweenExpr {
+    fn bind<const STRICT: bool>(
+        &self,
+        args: Vec<Box<dyn EvalExpr>>,
+    ) -> Result<Box<dyn EvalExpr>, BindError> {
+        let types = [TYPE_ANY, TYPE_ANY, TYPE_ANY];
```

Comment:
Similar to comparison ops, between should fail type check if t1 is not comparable to t2 or t1 is not comparable to t3 (thus resulting in an error in strict mode, missing in permissive mode). Perhaps can track as separate effort to fix and add conformance tests.

Reply:
#417